xtask/src/ci.rs

```diff
@@ -289,16 +289,11 @@ impl CiTask {
         .map_err(Into::into)
     }
 
-    /// Lint ruma-common with clippy with the nightly version and wasm target.
-    ///
-    /// ruma-common is currently the only crate with wasm-specific code. If that changes, this
-    /// method should be updated.
+    /// Lint ruma with clippy with the nightly version and wasm target.
     fn clippy_wasm(&self) -> Result<()> {
         cmd!(
-            "
-            rustup run {NIGHTLY} cargo clippy --target wasm32-unknown-unknown
-                -p ruma-common --features api,js,rand
-            "
+            "CLIPPY_CONF_DIR=.wasm rustup run nightly cargo clippy --target wasm32-unknown-unknown
+            -p ruma --features api,js,rand,canonical-json,events,__unstable-mscs"
```

Comment:
For this to work:

```suggestion
            "rustup run {NIGHTLY} cargo clippy --target wasm32-unknown-unknown
            -p ruma --features api,js,rand,canonical-json,events,__unstable-mscs"
        )
        .env("CLIPPY_CONF_DIR", ".wasm")
```

Reply:
Actually, to close #1875 I would add those features as well: `client-api,identity-service-api,signatures,markdown,html-matrix`

Reply:
This last comment hasn't been applied (sorry I should probably have edited the suggestion in the first comment).

Reply:
Ah i forgot , sorry about that! adding now

Reply:
Do we want `cargo clippy -- -D warnings` here?

Reply:
Yes, that's probably good.